language/bytecode-verifier/src/abstract_state.rs

```diff
@@ -4,98 +4,126 @@
 //! This module defines the abstract state for the type and memory safety analysis.
 use crate::{
     absint::{AbstractDomain, JoinResult},
+    borrow_graph::BorrowGraph,
     nonce::Nonce,
-    partition::Partition,
 };
-use mirai_annotations::checked_verify;
+use mirai_annotations::{checked_postcondition, checked_precondition, checked_verify};
 use std::collections::{BTreeMap, BTreeSet};
-use vm::file_format::{FieldDefinitionIndex, Kind, LocalIndex, StructDefinitionIndex};
+use vm::{
+    file_format::{
+        CompiledModule, FieldDefinitionIndex, Kind, LocalIndex, SignatureToken,
+        StructDefinitionIndex,
+    },
+    views::{FunctionDefinitionView, ViewInternals},
+};
 
 #[derive(Clone, Debug, Eq, PartialEq)]
+pub struct AbstractType {
```

Comment:
I like this name more than the old one of `StackAbstractValue`, but it feels a bit weird for the `AbstractType` to contain a value

Reply:
The value inside is "abstract" if that makes you feel better.

Reply:
It doesn't... the reason it makes me uncomfortable is it feels like a type (possibly abstract) paired with an abstract value. So its a (value, type), but it is named AbstractType. 

In short, I would not expect something named `AbstractType` to contain a value

Reply:
Renamed to TypedAbstractValue.